base/system/winlogon/sas.c

```diff
@@ -1066,7 +1068,7 @@ DoGenericAction(
             {
                 if (!Session->Gina.Functions.WlxIsLogoffOk(Session->Gina.Context))
                     break;
-                if (!NT_SUCCESS(HandleLogoff(Session, EWX_LOGOFF)))
+                if (!NT_SUCCESS(HandleLogoff(Session, wlxAction)))
```

Comment:
To understand a little bit more what's happening here, this change is in order to know whether this is a logoff because the user required a logoff, or a logoff because we are shutting down?